### ISSUE TITLE
建议古文从上至下书写，从左至右阅读

### ISSUE THREAD

**Issue by huangyz0918** [LingDong-/wenyan-lang#9]
> 古代的阅读顺序一开始是和竹简相关的，从上到下，从右到左，刚好符合竹简打开的顺序。

**Comment by minexo79** [comment]
那這樣的話可能連IDE都要重新寫一個喔

**Comment by yxliang01** [comment]
@huangyz0918 你的内容和标题不符诶 （从左到右 -> 从右到左）？

**Comment by huangyz0918** [comment]
@yxliang01 :P

**Comment by LingDong-** [comment]
Hi there,

Good suggestion! I'm thinking about doing that for the online IDE, perhaps as a toggle-able mode. But of course, as minexo79 mentioned, some re-writing needs to take place. 

My current priority is writing a language specification, but I'll gladly explore the vertical text orientation idea when I have time. Will update here when it's done!

Thanks.

**Comment by anymost** [comment]
Good idea which is deserved to try.

**Comment by jayhua0426** [comment]
有道理，提高逼格

**Comment by ZhenxingXiao** [comment]
> Hi there,
> 
> Good suggestion! I'm thinking about doing that for the online IDE, perhaps as a toggle-able mode. But of course, as minexo79 mentioned, some re-writing needs to take place.
> 
> My current priority is writing a language specification, but I'll gladly explore the vertical text orientation idea when I have time. Will update here when it's done!
> 
> Thanks.

Hi,
Do you think it is a good idea if we create a plugin of vs code to support this language. maybe the editor can implement the vertical test orientation idea. it may show Classical Chinese better.

**Comment by LingDong-** [comment]
@ZhenxingXiao Sounds like an appealing idea! There is already a nice [VS code plugin](https://github.com/antfu/wenyan-lang-vscode) by @antfu , perhaps we could reach out to them to see if it would be feasible to add vertical text, and how we can collaborate to make that happen ;)
Thanks!

**Comment by heangfat** [comment]
> 
> 
> > Hi there,
> > Good suggestion! I'm thinking about doing that for the online IDE, perhaps as a toggle-able mode. But of course, as minexo79 mentioned, some re-writing needs to take place.
> > My current priority is writing a language specification, but I'll gladly explore the vertical text orientation idea when I have time. Will update here when it's done!
> > Thanks.
> 
> Hi,
> Do you think it is a good idea if we create a plugin of vs code to support this language. maybe the editor can implement the vertical test orientation idea. it may show Classical Chinese better.

@ZhenxingXiao You may take a look at the following **Vertical Editors**:
[EverEdit](http://www.everedit.cn/) for Windows
[Mou](http://25.io/mou/) for Mac
[縱式](https://www.pixiv.net/fanbox/creator/38123590) for Mac

**Comment by heangfat** [comment]
@ZhenxingXiao @LingDong- Those editors above are only for your reference. 
Actually I suggest you to make a new one. As far as I know, 中國科學院軟件研究所 used to realize pure vertical GUI using Qt, including all the elements in a window. You may want to contact them for this.
![QtV](https://user-images.githubusercontent.com/20491276/71152634-f2e58b80-2271-11ea-8706-abe0083b0947.jpg)
Of course, this should be a long-term plan.

**Comment by NoDocCat** [comment]
强烈建议IDE字体改为<北师大说文小篆>
![image](https://user-images.githubusercontent.com/20502666/71173908-e1ff3f00-229e-11ea-892f-f6add67a6c97.png)

**Comment by heangfat** [comment]
@NoDocCat 北師大說文小篆很多字都顯示不出。譬如我記得「十」字便顯示爲空，查字型中有此字模，然而編碼映射有誤。此字型需要將編碼轉成萬國碼

**Comment by ZhenxingXiao** [comment]
> @ZhenxingXiao Sounds like an appealing idea! There is already a nice [VS code plugin](https://github.com/antfu/wenyan-lang-vscode) by @antfu , perhaps we could reach out to them to see if it would be feasible to add vertical text, and how we can collaborate to make that happen ;)
> Thanks!

Great! it will may look more like chinese classical book. 😁I will try to research that, and I will give a PR if I can figure it out.

**Comment by ZhenxingXiao** [comment]
> > > Hi there,
> > > Good suggestion! I'm thinking about doing that for the online IDE, perhaps as a toggle-able mode. But of course, as minexo79 mentioned, some re-writing needs to take place.
> > > My current priority is writing a language specification, but I'll gladly explore the vertical text orientation idea when I have time. Will update here when it's done!
> > > Thanks.
> > 
> > 
> > Hi,
> > Do you think it is a good idea if we create a plugin of vs code to support this language. maybe the editor can implement the vertical test orientation idea. it may show Classical Chinese better.
> 
> @ZhenxingXiao You may take a look at the following **Vertical Editors**:
> [EverEdit](http://www.everedit.cn/) for Windows
> [Mou](http://25.io/mou/) for Mac
> [縱式](https://www.pixiv.net/fanbox/creator/38123590) for Mac

Thanks, that's a valuable suggestion!

**Comment by ZhenxingXiao** [comment]
> @ZhenxingXiao @LingDong- Those editors above are only for your reference.
> Actually I suggest you to make a new one. As far as I know, 中國科學院軟件研究所 used to realize pure vertical GUI using Qt, including all the elements in a window. You may want to contact them for this.
> ![QtV](https://user-images.githubusercontent.com/20491276/71152634-f2e58b80-2271-11ea-8706-abe0083b0947.jpg)
> Of course, this should be a long-term plan.

yeah, it is a great idea. but there will be lots of work to do. maybe we can create a special editor base on node and electron. and we also can overwrite the render module in C/C++ if it is necessary.

**Comment by heangfat** [comment]
![直排界面](https://user-images.githubusercontent.com/20491276/71304211-98c60100-23fe-11ea-878b-e1863d4c0b1b.png)

縱排界面示例

**Comment by kotritrona** [comment]
如果是那个[Web IDE](http://wenyan-lang.lingdong.works/ide.html)的话，实现竖排很简单的，加上这两条css就可以了
```css
writing-mode: vertical-rl;
text-orientation: upright;
```

效果：https://i.imgur.com/UzK8Skq.png

**Comment by UltimatePea** [comment]
我在向vscode官方团队建议加入这个功能，大家请帮忙个投票（点击大拇指👍图标），谢谢！链接：https://github.com/microsoft/vscode/issues/152005
===
I submitted a feature to vscode for adding this feature. Please upvote the issue by clicking the thumbs up 👍 button. Thank you! Link: https://github.com/microsoft/vscode/issues/152005
